### ISSUE TITLE
Доработка симулятора.

### ISSUE THREAD

**Issue by zelezomen** [SBProX/SBProX-Dev#269]
1) Изменения цены происходят не по сделкам, а по какому-то другому принципу, динамика не та, такое ощущение, что тики проредили раза в три. 
2) Нужно добавить поле "Start time" для того, чтобы торгующие америку не ждали пока чарт дойдёт до нужного времени. Рядом с полем кнопку перехода на это время, чтобы она проматывала чарт до ближайшего указанного значения. Например start time 9:00, если сейчас в симуляторе 6:00, то перемотает на 3 часа, если сейчас 10:00, то на 9:00 следующей сессии. 
3) Нужно добавить перемотку. Поле с выбором шага (1m, 2m, 5m, 15, и т.д. до 4H) и кнопку.
4) С торговым функционалом нужно что-то делать полностью, так, как есть - не годится совсем. 
- не видно, что сейчас выбрано - лимит или стоп или ещё что;
- несколько входов суммируются в одну среднюю цену, не самое удобное решение, нужно сделать выбор;
- несколько ордеров на одной цене просто перекрываются, не видно сколько их там;
- плашки ордеров пляшут в размере при перемещении, зависимо от наличия дробного значения (длинны цифры цены), после нескольких ордеров в глазах рябит;
- не хватает линии цены ордера на всю длину поля чарта, хотя бы в момент выставления/изменения;
- не хватает хоткеев. 

Не вижу ничего зазорного в том, чтобы взять идеи или решения из других платформ. Из всего, что пробовал, самый удобный вариант в волфиксе - аккуратно, удобно и функционально.

**Comment by finast** [comment]
Поддержу. Стоит улучшить этот нужный функционал.

**Comment by bushevigor** [comment]
ПОДДЕРЖИВАЮ ОБЕИМИ РУКАМИ

**Comment by Flawmaster** [comment]
Поддержу, но если учесть темы, что открывали до этого, то тут, может быть, 1 пункт останется.

**Comment by bizlis** [comment]
Добавлю для доработки симулятора: загрузку истории надо сделать по другому принципу, не так сейчас реализовано (хотя это конечно лучше чем ничего), опишу принцип работы подгрузки необходимого периода истории в симуляторе, как я вижу для оптимальной работы тестирования стратегий:

Пользователь открывает необходимое количество ТФ (к примеру Д1, Н1, М5, М1)
На каждом ТФ пользователь устанавливает свой период подгрузки графика (к примеру для Д1=год, Н1=3 месяца, М5=неделя, М1=день). Т.е. конечная дата на всех ТФ одна и та же, а начальная дата графика разная.
На всех ТФ настраиваются графики, добавляются индикаторы
Выбирается режим симулятор и устанавливается начальная дата, с которого начнется тестирование.

В итоге мы имеем некоторую подгруженную историю, с необходимыми настройками, на которой проводится предварительный анализ, а запуск теста текущих торгов дает отработку стратегии.

Суть в том, что необходимо увидеть как движется цена относительно дня, недели, месяца одновременно, для этого и выбираются разные начальные периоды истории для разных ТФ.

**Comment by SBProX** [comment]
1. Вы ошибаетесь. Вы устанавливаете скорость - кол-во тиков.
Например, если вы ставите 100, то это значит что симулятор будет отбирать 100 тиков в 1 секунду.
А как там эти тики формировались на рынке, мы это не можем контролировать.

По остальным пунктам постараемся к вам прислушаться, но ничего пока не обещаю.

**Comment by zelezomen** [comment]
> 1. Вы ошибаетесь. Вы устанавливаете скорость - кол-во тиков.
>    Например, если вы ставите 100, то это значит что симулятор будет отбирать 100 тиков в 1 секунду.
>    А как там эти тики формировались на рынке, мы это не можем контролировать.
> 
> По остальным пунктам постараемся к вам прислушаться, но ничего пока не обещаю.
Нет, не ошибаюсь. Речь идёт о скорости 1х. Вы верно подметили, происходит именно одно изменение в секунду (или около того), что не отражает действительность. На реальном рынке и тренажёре, которым пользуюсь сейчас, я вижу все колебания цены именно в той динамике в которой они происходят или происходили. Это важно в моменте, т.к. является частью информации о цене- как проходим возможную зону ликвидности, где провалы, где агрессия, где слабость и т.д.. В случае с сбпроХ динамика всё время одна - раз в секунду, и не понятно, что сейчас происходит - цену агрессивно зажали и будет прострел или в инструменте вообще никого нет - характер движения один и тот же. При торговле просто в ступор входишь - судя по объёмам в инструменте жесть, а динамика изменения цены как ни в чём не бывало, сначала даже не понял в чём проблема.

**Comment by SergeyLos** [comment]
> 3. Нужно добавить перемотку. Поле с выбором шага (1m, 2m, 5m, 15, и т.д. до 4H) и кнопку.

Поддерживаю по поводу поля с выбором шага, т.к. есть те кто торгует на Н1 и не обязательно ждать n-ое количество времени пока бар сформируется. Было бы удобно выбрать дискретный шаг из стандартных ТФ и нажимая например пробел смотреть как у тебя меняются (при ТФ ниже текущего ТФ) или идут бары (при ТФ равным текущему ТФ).

Дополнительно было бы удобно иметь несколько окон с разными ТФ. Было бы вообще супер, если разные окна можно будет перетаскивать на разные мониторы.

Заранее спасибо.

PS: как говорилось ранее не зазорно взять что-то из других платформ, например ForexTester :).

**Comment by Vegas3737** [comment]
Поддерживаю Решение , по примеру платформы Tiger Trade . Друзья просто возьмите и скопируйте от туда весь интерфейс !! просто вы взгляните туда какой у них тестр я лучше не видел , стоит 1000р/мес
